opennlp-tools/src/main/java/opennlp/tools/util/TrainingParameters.java

```diff
@@ -37,26 +37,50 @@
   public static final String CUTOFF_PARAM = "Cutoff";
   public static final String THREADS_PARAM = "Threads";
 
-  private Map<String, String> parameters = new HashMap<>();
+  private Map<String, Object> parameters = new HashMap<>();
 
   public TrainingParameters() {
   }
 
   public TrainingParameters(TrainingParameters trainingParameters) {
     this.parameters.putAll(trainingParameters.parameters);
   }
-  
+
   public TrainingParameters(Map<String,String> map) {
-    parameters.putAll(map);
+    //parameters.putAll(map);
+    // try to respect their original type...
+    for (String key: map.keySet()) {
```

Comment:
Is there anyway to avoid this type of try/catch structure? Just kind of thinking out loud here. Could you have classes that extend an `AbstractTrainingParameter` like `IntegerTrainingParameter`, `StringTrainingParameter`, etc. in order to know the type of value that it contains? `TrainingParameters` would then be a list of `AbstractTrainingParameter`.